packages/api/src/apps/api/README.md

```diff
@@ -117,6 +117,10 @@ that call uses the collateral address, not the emp address.
 
 Get only tvl (and other future statistics) for a single emp.
 
-#### listEmpStats(currency: "usd" = "usd") => StatData[]
+### listEmpStats(currency: "usd" = "usd") => StatData[]
 
 Get all tvl (and other future statistics) for all known emps.
+
+### tvl(addresses?: string[], currency: CurrencySymbol = "usd") => string
+
+Returns tvl of all supplied addresses in USD 18 decimals. If no addresses supplied, returns TVL of all known contracts.
```

Comment:
can you explain what the process is of new contracts becoming "known"? is it about a contract being requested at some point and then it is cached into the future?

Reply:
This is all based off the registry contract. Theres a service that pools the contract for events and will add emp addresses as they get added, so technically this will be handled in somewhat real time. 

You can see [here](https://github.com/UMAprotocol/protocol/blob/master/packages/api/src/services/emp-registry.ts#L18), the update function on the registry service simply just adds emp addresses to the set of known emps.

Reply:
With the OO, what happens when contracts aren't registered with the Registry? For instance, the LSP will not be registered.

Reply:
The registry service handles adding emps from registry events, if there are new types of contracts, i would just add another service which specializes in those contracts, however they are discovered.  I might need to be sold a bit on the factory approach, i dont immediately see the value of it, but it sounds like something suited more in the SDK.

for one offs or custom addresses, this could be handled through priveleged calls in the api, which adds addresses to a table, or through envs/configs, CMS or git repo. what ever makes the most sense.